src/Compilers/Core/Portable/Symbols/ISymbolExtensions_IsAccessibleWithin.cs

```diff
@@ -0,0 +1,348 @@
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            bool sameOriginalNamedType(INamedTypeSymbol t1, INamedTypeSymbol t2)
+            {
+                Debug.Assert(t1 != null);
+                Debug.Assert(t1.IsDefinition);
+                Debug.Assert(t2?.IsDefinition != false);
+
+                // We expect a given named type definition to satisfy reference identity.
+                // But we relax this to permit a type to be represented by multiple symbols (e.g. separate compilations).
+                // Note that the same symbol is expected to have the identical name as itself, despite VB language rules.
+                return t1 == t2 || t2 != null && t1.Name == t2.Name && t1.Arity == t2.Arity && sameOriginalSymbol(t1.ContainingSymbol, t2.ContainingSymbol);
+
+                bool sameOriginalSymbol(ISymbol s1, ISymbol s2)
+                {
+                    if (s1 == s2)
+                    {
+                        return true;
+                    }
+
+                    if (s1 == null || s2 == null)
+                    {
+                        return false;
+                    }
+
+                    switch (s1.Kind)
+                    {
+                        case SymbolKind.NamedType:
+                            return sameOriginalNamedType(s1 as INamedTypeSymbol, s2 as INamedTypeSymbol);
+                        case SymbolKind.Namespace:
+                            return sameNamespace(s1 as INamespaceSymbol, s2 as INamespaceSymbol);
+                        case SymbolKind.Assembly:
+                            return sameAssembly(s1 as IAssemblySymbol, s2 as IAssemblySymbol);
+                        case SymbolKind.NetModule:
+                            return sameModule(s1 as IModuleSymbol, s2 as IModuleSymbol);
+                        default:
+                            return false;
+                    }
+                }
+
+                bool sameModule(IModuleSymbol m1, IModuleSymbol m2)
+                {
+                    Debug.Assert(m1 != null);
+                    // We don't need to check the module name, as modules are effectively merged in the containing assembly.
+                    return m2 != null && sameOriginalSymbol(m1.ContainingSymbol, m2.ContainingSymbol);
+                }
+
+                bool sameAssembly(IAssemblySymbol a1, IAssemblySymbol a2)
+                {
+                    Debug.Assert(a1 != null);
+                    return a2 != null && a1.Identity.Equals(a2.Identity);
+                }
+
+                bool sameNamespace(INamespaceSymbol n1, INamespaceSymbol n2)
+                {
+                    Debug.Assert(n1 != null);
+                    // Note that the same symbol is expected to have the identical name as itself, despite VB language rules.
```

Comment:
Your comment is too vague to be actionable. Please identify specific scenarios that you do not think will work well for cross-language queries. Please identify specific scenarios that you don't think will work well for VB to VB comparison across different compilations. Please identify specific scenarios that you believe might probably produce false positives.

Reply:
> Your comment is too vague to be actionable. Please identify specific scenarios that you do not think will work well for cross-language queries. Please identify specific scenarios that you don't think will work well for VB to VB comparison across different compilations. Please identify specific scenarios that you believe might probably produce false positives.

Clearly you already identified the inaccuracy of this check, hence the comment. VB namespaces are merged case-insensitively, it is obvious that the name of the namespace doesn't always reflect the actual casing of the name, therefore this comparison could produce both false negatives and false positives as the result. Since you are the author of the change, I leave it to you to come up with relevant unit-tests to demonstrate that. If you have trouble with those, I can help you offline with constructing scenarios.  

---
In reply to: [185161899](https://github.com/dotnet/roslyn/pull/26367#discussion_r185161899) [](ancestors = 185161899)

Reply:
Added a doc comment reflecting that, a test showing that, and filed an issue for the VB compiler not obeying the contract of `MetadataName`.

---
In reply to: [185126619](https://github.com/dotnet/roslyn/pull/26367#discussion_r185126619) [](ancestors = 185126619)